javatools/jarutil.py

```diff
@@ -85,13 +85,25 @@ def verify(certificate, jar_file):
 
     if len(sf_files) == 0:
         raise JarSignatureMissingError("No .SF file in %s" % jar_file)
-
     elif len(sf_files) > 1:
-        # This is acceptable: SF file represents a signer. But then the
-        # validation logic becomes more complicated...
-        msg = "Multiple .SF files in %s, this is not supported yet" % jar_file
-        raise VerificationError(msg)
-
+        if sf_name is None:
+            msg = "Multiple .SF files in %s, but SF_NAME.SF not specified" \
+                    % jar_file
+            raise VerificationError(msg)
+        elif ('META-INF/' + sf_name) in sf_files:
+            sf_filename = 'META-INF/' + sf_name
+        else:
+            msg = "No .SF file in %s named META-INF/%s (found %d .SF files)" \
+                    % (jar_file, sf_name, len(sf_files))
+            raise VerificationError(msg)
+    elif len(sf_files) == 1:
+        if sf_name is None:
+            sf_filename = sf_files[0]
+        elif sf_files[0] == 'META-INF/' + sf_name:
+            sf_filename = sf_files[0]
+        else:
+            msg = "No .SF file in %s named META-INF/%s" % (jar_file, sf_name)
+            raise VerificationError(msg)
     sf_filename = sf_files[0]
```

Comment:
But we might have set `sf_filename` to some other `sf_file` earlier, e.g. at line 94?